diff -Nur graph500-2.1.4/make.inc graph500-2.1.4.patched/make.inc
--- graph500-2.1.4/make.inc	1970-01-01 05:30:00.000000000 +0530
+++ graph500-2.1.4.patched/make.inc	2015-10-11 16:40:26.942115028 +0530
@@ -0,0 +1,10 @@
+# -*- Makefile -*-
+# Copyright 2010, Georgia Institute of Technology, USA.
+# See COPYING for license.
+CFLAGS = -g -std=c99
+#CFLAGS = -g -std=c99 -O3 -march=native -fgcse-sm -fgcse-las -fgcse-after-reload -floop-strip-mine -ftree-loop-im -fivopts -funswitch-loops
+LDLIBS = -lm -lrt
+CPPFLAGS += #-DUSE_MMAP_LARGE -DUSE_MMAP_LARGE_EXT
+
+BUILD_OPENMP = Yes
+CFLAGS_OPENMP = -fopenmp
